src/highlight/HighlightListener.ts

```diff
@@ -1,97 +1,36 @@
 import debounce from 'lodash/debounce';
-import {
-    AppStore,
-    getIsInitialized,
-    getIsSelecting,
-    SelectionArg as Selection,
-    setIsSelectingAction,
-    setSelectionAction,
-} from '../store';
-import { MOUSE_PRIMARY } from '../constants';
+import { AppStore, getIsSelecting, SelectionArg as Selection, setSelectionAction } from '../store';
 
 export type Options = {
     getSelection: () => Selection | null;
-    selectionChangeDelay?: number;
     store: AppStore;
 };
 
 // Debounce 500ms for keyboard selection
 const SELECTION_CHANGE_DEBOUNCE = 500;
 
 export default class HighlightListener {
-    annotatedEl?: HTMLElement;
-
     getSelection: () => Selection | null;
 
-    selectionChangeDelay?: number;
-
-    selectionChangeTimer?: number;
-
     store: AppStore;
 
-    constructor({ getSelection, selectionChangeDelay = 0, store }: Options) {
+    constructor({ getSelection, store }: Options) {
         this.getSelection = getSelection;
-        this.selectionChangeDelay = selectionChangeDelay;
         this.store = store;
 
         document.addEventListener('selectionchange', this.debounceHandleSelectionChange);
```

Comment:
Should we also move `selectionchange` logic to the new component so that we can remove HighlightListener?

Reply:
I don't think so, I left it this way so that there is one `selectionchange` listener on the document -- if we moved this into `HighlightCreatorManager` then there would be one `selectionchange` listener added to the document per page